Lib/fontTools/varLib/instancer/__init__.py

```diff
@@ -1326,29 +1340,27 @@ def parseLimits(limits: Iterable[str]) -> Dict[str, Optional[AxisTriple]]:
     result = {}
     for limitString in limits:
         match = re.match(
-            r"^(\w{1,4})=(?:(drop)|(?:([^:]+)(?:[:]([^:]+))?(?:[:]([^:]+))?))$",
+            r"^(\w{1,4})=(?:(drop)|(?:([^:]*)(?:[:]([^:]*))?(?:[:]([^:]*))?))$",
             limitString,
         )
         if not match:
             raise ValueError("invalid location format: %r" % limitString)
         tag = match.group(1).ljust(4)
+
         if match.group(2):  # 'drop'
-            lbound = None
-        else:
-            lbound = strToFixedToFloat(match.group(3), precisionBits=16)
-        ubound = default = lbound
-        if match.group(4):
-            ubound = default = strToFixedToFloat(match.group(4), precisionBits=16)
-            default = None
-        if match.group(5):
-            default = ubound
-            ubound = strToFixedToFloat(match.group(5), precisionBits=16)
-
-        if all(v is None for v in (lbound, default, ubound)):
             result[tag] = None
             continue
 
-        result[tag] = AxisTriple(lbound, default, ubound)
+        triple = match.group(3, 4, 5)
+
+        if triple[1] is None:  # "value" syntax
+            triple = (triple[0], triple[0], triple[0])
+        elif triple[2] is None:  # "min:max" syntax
+            triple = (triple[0], None, triple[1])
```

Comment:
this `if` is not immediately obvious to me..
maybe can you also add tests for the argument parsing?

Reply:
ok I did https://github.com/fonttools/fonttools/pull/3323/commits/27c79679eb947a22a2ee79a60526cb0f95b9376b